client_apis/python/src/tests/integration/user_test.py

```diff
@@ -0,0 +1,252 @@
+        user_data = self.TestDataGen.gen_user()
+
+        user_add_params = self._convert_user_data_to_user_add_params(user_data)
+        user_add_params['confirm_password'] = "wr0ngp@ssw0rd"
+
+        # should not be able to create user with bad password confirmation
+        with self.assertRaises(requests.HTTPError) as cm:
+            cb.user_add_from_data(**user_add_params)
+
+        self.assertEqual(cm.exception.response.status_code, 400)
+
+    def test_create_valid_user(self):
+        self._test_create_valid_user()
+
+    def test_create_valid_global_admin(self):
+        self._test_create_valid_global_admin()
+
+    def test_create_user_duplicate_username(self):
+        first_user = self._test_create_valid_user()
+        self.assertIn('username', first_user)
+        username = first_user['username']
+
+        # generate second user info, but then use the same username as the first user
+        second_user_data = self.TestDataGen.gen_user()
+        second_user_data['username'] = username
+        second_user_add_params = self._convert_user_data_to_user_add_params(second_user_data)
+
+        # attempt create user with the same username
+        with self.assertRaises(requests.HTTPError) as cm:
+            cb.user_add_from_data(**second_user_add_params)
+
+        self.assertEqual(cm.exception.response.status_code, 409)
+
+    ## Delete user tests
+
+    def test_delete_unknown_user(self):
+        username = "unknown-user"
+
+        # verify the user does not exist on the server
+        self._assert_user_doesnt_exist(username)
+
+        # attempt to delete the user
+        with self.assertRaises(requests.HTTPError) as cm:
+            cb.user_del(username)
+
+        self.assertEqual(cm.exception.response.status_code, 500)
+
+    def test_delete_new_user(self):
+        # create the user
+        new_user = self._test_create_valid_user()
+        self.assertIn('username', new_user)
+        username = new_user['username']
+
+        # verify the user exists on server
+        retrieved_user = cb.user_info(username)
+        self._verify_retrieved_user(retrieved_user)
+
+        # delete the user
+        cb.user_del(username)
+
+        # verify the user no longer exists on the server
+        retrieved_user = cb.user_info(username)
+        self.assertIsNone(retrieved_user)
+
+    def test_delete_new_global_admin(self):
+        # create the global admin user
+        new_user = self._test_create_valid_global_admin()
+        self.assertIn('username', new_user)
+        username = new_user['username']
+
+        # verify the user exists on server
+        retrieved_user = cb.user_info(username)
+        self._verify_retrieved_user(retrieved_user)
+
+        # delete the user
+        cb.user_del(username)
+
+        # verify the user no longer exists on the server
+        self._assert_user_doesnt_exist(username)
+
+    ## Test Helpers
+
+    def _test_create_valid_user(self):
```

Comment:
Do we need to include `test` in the private method names? The public methods are prefixed with `test` because that is how `unittest` distinguishes between tests and non-tests.